geoagent/tools/images.py

```
@@ -102,13 +128,240 @@ def _generate_openai_image(
+    endpoint: str, api_key: str, payload: dict[str, Any], timeout: float
+) -> bytes:
+    """POST a JSON payload to the MiniMax image endpoint and return the body."""
+    request = urllib.request.Request(
+        endpoint,
+        data=json.dumps(payload).encode("utf-8"),
+        method="POST",
+        headers={
+            "Authorization": f"Bearer {api_key}",
+            "Content-Type": "application/json",
+        },
+    )
```

Ruff (0.16.0)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)